etl/src/main/java/org/pih/hivmigration/etl/sql/VitalsMigrator.groovy

```diff
@@ -0,0 +1,85 @@
+package org.pih.hivmigration.etl.sql
+
+class VitalsMigrator extends SqlMigrator {
+    @Override
+    def void migrate() {
+
+        executeMysql("Create vitals staging table", '''
+            CREATE TABLE hivmigration_vitals (
+              source_encounter_id int,
+              sign varchar(30),
+              result double,
+              result_unit varchar(100),
+              source_patient_id int,
+              KEY `source_encounter_idx` (`source_encounter_id`)
+            );
+        ''')
+
+        // TODO: should we do something with the 563 vitals with encounter type 'food_study'
+        //      or the 3 with type patient_contact?
+        loadFromOracleToMySql('''
+            INSERT INTO hivmigration_vitals (source_encounter_id, sign, result, result_unit, source_patient_id)
+            VALUES (?, ?, ?, ?, ?)
+        ''', '''
+            select 
+                v.ENCOUNTER_ID, 
+                v.SIGN, 
+                v.RESULT, 
+                v.RESULT_UNIT,
+                e.PATIENT_ID
+            from HIV_EXAM_VITAL_SIGNS v, HIV_ENCOUNTERS e
+            where v.ENCOUNTER_ID = e.ENCOUNTER_ID 
+            and SIGN in ('height', 'weight', 'bmi')
+            and e.TYPE in ('intake', 'followup')
+            and e.PATIENT_ID is not null
+        ''')
+
+        executeMysql("CALL create_tmp_obs_table();")
+
+        setAutoIncrement("tmp_obs", "(SELECT max(obs_id) + 1 FROM obs)")
+
+        executeMysql("Add vitals obs", '''
+            INSERT INTO tmp_obs (source_patient_id, source_encounter_id, value_numeric, concept_uuid)
+            SELECT
+                source_patient_id,
+                source_encounter_id,
+                CASE
+                    -- result unit is unreliable, but fortunately the values are unambiguous
+                    WHEN result < 3.0 THEN round(result * 100, 1)
+                    WHEN result > 40 AND result < 300 THEN result
+                END,
+                '3ce93cf2-26fe-102b-80cb-0017a47871b2'
+            FROM hivmigration_vitals
+            WHERE sign = 'height'
+            AND (result < 3.0 OR (result > 40 AND result < 300));
+
+            INSERT INTO tmp_obs (source_patient_id, source_encounter_id, value_numeric, concept_uuid)
+            SELECT
+                source_patient_id,
+                source_encounter_id,
+                CASE
+                    WHEN result_unit = 'kgs' THEN result
+                    WHEN result_unit = 'lbs' THEN round(result * 0.453592, 1)
+                    -- Drop 3461 entries with result_unit = NULL
```

Comment:
No, we're not dropping values.  We should migrate these in and highlight that these need to be looked at from a data quality perspective and validated.  It is likely that we can determine the correct units based on comparison with other data in the patient record.  What other weights did the patient have before/after?  Look at whether the value or converted value falls into the same range.  Or migrate as is in KG and leave this for a data cleaning exercise.  Don't discard the data.

Reply:
Sounds like @mseaton  will disagree but I'd be okay with dropping these for now if we provide a report of the bad values to the team on the ground and have a follow up ticket for deciding what we want to do about them.

Reply:
Unless these values are causing problems, my preference would be to include them...and then ticket as something to review and ensure are done properly, rather than to exclude them and then ticket.

Reply:
Migrating as kg and ticketing seems reasonable to me. I wonder if we should figure out a way to make the pipeline produce a log file of data to review—something we could pass off to the team that's like

| patient | field | value |
|-------|----|----|
| 12345 | height | 472cm |
| 5423 | birthdate | 1800-01-01 |
...

Reply:
That's cool thought @brandones , having the pipeline produce a table/log of data to be validated.